spec/index.html

```diff
@@ -0,0 +1,1272 @@
+        1. Else,
+          1. Let _prototype_ be %Function.prototype%.
+        1. Let _sourceText_ be the source text matched by |MethodDefinition|.
+        1. Let _closure_ be OrdinaryFunctionCreate(_prototype_, _sourceText_, |UniqueFormalParameters|, |FunctionBody|, ~non-lexical-this~, _env_, _privateEnv_).
+        1. Perform MakeMethod(_closure_, _object_).
+        1. <ins>If _object_ has a [[StructBrand]] internal slot, then</ins>
+          1. <ins>NOTE: Struct instance methods' home object have a [[StructBrand]] internal slot.</ins>
+          1. <ins>Set _closure_ to ? StructMethodCreate(_closure_).</ins>
+        1. Return the Record { [[Key]]: _propKey_, [[Closure]]: _closure_ }.
+      </emu-alg>
+    </emu-clause>
+  </emu-clause>
+
+  <emu-clause id="sec-changes-to-modules">
+    <h1>Changes to Modules</h1>
+
+    <emu-grammar type="definition">
+      ExportDeclaration :
+        `export` ExportFromClause FromClause `;`
+        `export` NamedExports `;`
+        `export` VariableStatement[~Yield, +Await]
+        `export` Declaration[~Yield, +Await]
+        `export` `default` HoistableDeclaration[~Yield, +Await, +Default]
+        `export` `default` ClassDeclaration[~Yield, +Await, +Default]
+        <ins>`export` `default` StructDeclaration[~Yield, +Await, +Default]</ins>
+        `export` `default` [lookahead &notin; { `function`, `async` [no LineTerminator here] `function`, `class` }] AssignmentExpression[+In, ~Yield, +Await] `;`
+    </emu-grammar>
+
+    <emu-clause id="sec-exports-runtime-semantics-evaluation" type="sdo">
+      <h1>Runtime Semantics: Evaluation</h1>
+      <ins class="block">
+      <emu-grammar>ExportDeclaration : `export` `default` StructDeclaration</emu-grammar>
+      <emu-alg>
+        1. Let _value_ be ? BindingStructDeclarationEvaluation of |StructDeclaration|.
+        1. Let _structName_ be the sole element of the BoundNames of |StructDeclaration|.
+        1. If _structName_ is *"\*default\*"*, then
+          1. Let _env_ be the running execution context's LexicalEnvironment.
+          1. Perform ? InitializeBoundName(*"\*default\*"*, _value_, _env_).
+        1. Return ~empty~.
+      </emu-alg>
+      </ins>
+    </emu-clause>
+  </emu-clause>
+</emu-clause>
+
+<emu-clause id="sec-shared-structs">
+  <h1>Shared Structs</h1>
+
+  <emu-clause id="sec-shared-struct-exotic-objects">
+    <h1>Shared Struct Exotic Objects</h1>
+    <ins class="block">
+    <p>Shared Structs are fixed-layout exotic objects that can be shared across agents and be accessed in parallel from multiple agents. They are like structs with more restricted behaviour so as to be possible to be shared across agents. They cannot contain methods or private fields. Their fields can only hold primitives or other shared values. Accessing their fields is unordered by default and is governed by the memory model. Such accesses can be made sequentially consistent by using newly overloaded Atomics methods.</p>
```

Comment:
> They cannot contain methods or private fields.

Given shared structs will be able to be correlated by source position anyway is the restriction on private fields really neccessary? It seems unfortunate that things like shared queues can't encapsulate their internals.

Reply:
Agreed, that'd be nice, but this proposal is already quite large, so one step at a time. If the correlation registry thing gets consensus, private fields is a sensible immediate follow-up proposal.